frontend/src/bundles/talent-onboarding/store/actions.ts

```diff
@@ -20,15 +20,27 @@ const updateTalentDetails = createAsyncThunk<
     UserDetailsGeneralCustom,
     AsyncThunkConfig
 >(`${sliceName}/update-talent-details`, async (updatePayload, { extra }) => {
-    const { talentOnBoardingApi } = extra;
+    const { talentOnBoardingApi, fileUploadApi } = extra;
+    const { cv, photo, ...restPayload } = updatePayload;
+
+    if (cv && photo) {
+        const { document, image } = await fileUploadApi.upload({
+            files: [cv, photo],
+        });
+
+        restPayload.photoId = image.id;
+        restPayload.cvId = document.id;
+    }
```

Comment:
Hello, for this I did it this way because step 4 of onboarding has validation on the form which refuses to submit if photo or cv is missing. The 'if' is only in place because in update request dto cv and photo are optional.